### ISSUE TITLE
gmpy2.mpq can not be initialized from a pair of objects defining `__mpz__`

### ISSUE THREAD

**Issue by vinklein** [aleaxit/gmpy#237]
`mpq` can be initialized with one object defining `__mpz__` but not with a pair:
```
>>> from gmpy2 import mpz
>>> import gmpy2
>>> class One:
...     def __mpz__(self): return mpz(1)
... 
>>> class Two:
...     def __mpz__(self): return mpz(2)
... 
>>> gmpy2.mpq(Two())
mpq(2,1)
>>> gmpy2.mpq(One(), Two())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: cannot convert object to mpq
```

This issue was first found by videlec in sage see [#27632](https://trac.sagemath.org/ticket/27632)

**Comment by casevh** [comment]
I will look at this tonight.

**Comment by casevh** [comment]
Quick look. There are some inconsistencies in the MPQ conversion functions defined in gmpy2-cache.c. MPQ_From_Rational doesn't handle the __mxx__ conversions but the MPQ_From_Number does. So it may be sufficient to change the two calls to From_Rational to From_Number. But the real issue is that only MPQ defines a From_Number function. We should either align MPQ with the other types or align the other types with MPQ.

I am fine with a quick fix for beta1 and standardize the conversion rules for beta2. But I am concerned that we will quick find more inconsistencies and need to do a quick beta2. Thoughts?

**Comment by vinklein** [comment]
Thanks for your answer. Using From_Number instead of From_Rational works.

**Comment by vinklein** [comment]
> We should either align MPQ with the other types or align the other types with MPQ.
I don't have a strong opinion on it but have noticed that `MPQ_From_Number`  is almost always used when `IS_RATIONAL(ourvalue)` is true. That means that the case for `float` and `mpfr` inside `MPQ_From_Number` have little interest.

**Comment by casevh** [comment]
Closed by PR #240.
